Refactor: Include `Template.default` when Querying Funder Templates in `TemplateOptionsController`

aaronskiba opened the pull request:
# Changes proposed in this PR:
Simplified template loading logic by removing explicit `Template.default`  handling.
- `Template.latest_customizable.where(org_id: funder.id)` and `Template.published.publicly_visible.where(org_id: funder.id, customization_of: nil)` now fetch the default template as well
- Removed manual insertion and filtering of the default template; `sort_templates()` now determines ordering.

aaronskiba committed:
Clean up comments in TemplateOptionsController

aaronskiba committed:
Refactor: Include default template in funder templates query

Simplified template loading logic by removing explicit default template handling.
- `Template.latest_customizable.where(org_id: funder.id)` and `Template.published.publicly_visible.where(org_id: funder.id, customization_of: nil)` now fetch the default template as well
- Removed manual insertion and filtering of the default template; `sort_templates()` now determines ordering.

momo3404 commented on app/controllers/template_options_controller.rb:
```diff
@@ -24,8 +24,8 @@ def index
-      # Load the funder's template(s) minus the default template (that gets swapped in below)
-      @templates = Template.latest_customizable.where(org_id: funder.id, is_default: false).sort_by(&:title).to_a
+      # Load the funder's template(s)
+      @templates = Template.latest_customizable.where(org_id: funder.id).sort_by(&:title).to_a
```
Why was `is_default` removed?

aaronskiba replied:
We want the funder's default template as well. Previously we were doing this in two steps -- the first was this query that fetched everything except for the default template. The second step fetched just the funder's default template, and then reordered the templates so the default template was at the beginning. Both steps also perform the same funder vs customization comparison. Since sort_templates() is now handling the reordering, it makes sense to just fetch all of the funder templates (default and non-default) together to start.

The following is the now removed code that handled the second step:
```ruby
  # Always use the default template
  if Template.default.present? && org.present?
    # Fetch the org’s latest customization of the default template
    customization = Template.published.latest_customized_version(Template.default.family_id, org.id).first
    # In short: check if `customization` is based on `Template.default`
    # If so, use `customization`; otherwise, fall back to `Template.default`
    customization = Template.default unless customization.present? && !customization.upgrade_customization?
    @templates.select! { |t| t.id != Template.default.id && t.id != customization.id }
    # We want the default template to appear at the beggining of the list
    @templates.unshift(customization)
  end
```

momo3404 commented on app/controllers/template_options_controller.rb:
```diff
@@ -46,22 +46,10 @@ def index
     else
       # if'No Primary Research Institution' checkbox is checked,
```
Do we have the `No Primary Research Institution` checkbox in DMP Assistant?

aaronskiba replied:
It must just be referring to the following:
<img width="1174" height="195" alt="Screenshot from 2025-10-06 15-57-24" src="https://github.com/user-attachments/assets/db2dd52f-d755-4128-ba89-3a9279507dca" />

momo3404 commented on app/controllers/template_options_controller.rb:
```diff
@@ -46,22 +46,10 @@ def index
     else
       # if'No Primary Research Institution' checkbox is checked,
       # only show publicly available template without customization
-      @templates << Template.default if Template.default.present?
```
Do you know if this line used to actually add to `templates`?

aaronskiba replied:
BEFORE
```ruby
@templates << Template.default if Template.default.present?
@templates << Template.published.publicly_visible.where(org_id: funder.id, customization_of: nil,
                                                              is_default: false).sort_by(&:title).to_a
```

AFTER
```ruby
@templates << Template.published.publicly_visible.where(org_id: funder.id,
                                                              customization_of: nil).sort_by(&:title).to_a
```

It was adding it before, and now it's adding it still. It was okay before, because at the time we did want `Template.default` to be the first template. Now that the ordering has changed, I think it makes sense to just fetch everything in a single DB query. Removing `is_default: false` should allow us to do that.

aaronskiba committed:
Fix test: Associate default template with default funder

The changes in a4fc95edf50f91f7647e3891948d28e7c2d9fc43 require that the default template belong to the default org.

momo3404 reviewed (approved):
Nice refactoring work, this file is a lot less convoluted now.
